libraries/app/database_api.cpp

```diff
@@ -1554,11 +1554,11 @@ vector<market_trade> database_api_impl::get_trade_history( const string& base,
          {
             trade.sequence = -itr->key.sequence;
             trade.side1_account_id = itr->op.account_id;
-            trade.side1_was_selling = itr->op.receives;
+            trade.side1_was_selling = itr->op.pays.asset_id(_db).symbol;
```

Comment:
Querying the database (the `id(db)` call) repeatedly for the same data inside a loop is inefficient.

By the way, what do you think is a better design (for usability, think as if you're a trader or a client app developer, and an API server admin):
1. return the "raw" data, e.g. add two fields `side1_was_selling:CNY` and `side2_was_selling:USD`,  or only add one of the two to save bandwidth;
2. "translate" the data, e.g. add one field and return `"trade_type":"buy"` or `"trade_type":"sell"` according to other known fields (base, quote, price, amount and value), or even shorter e.g. `"type":"buy"`. More info in the updated OP of the issue.

Reply:
It is a difficult question where actually all the options can be valid. As we already have `side1_account_id` and `side2_account_id` i will be expecting the data on what was sold to be there on both sides. We can change the field names to `side1_sold` and `side2_sold` to save a few chars, i am not against having it only in one side  but i dont prefer it.

Reply:
Imagine a market EUR:USD where `base = EUR` and `quote = USD`, when there was a trade, say, Alice sold 2 `EUR` to Bob at price `1.05 USD per EUR` (which may imply that Alice was the taker). The old returned data would look like:
```
price : 1.05
amount : 2
value : 2.1
side1 (maker) : bob
side2 (taker) : alice
```

Nevertheless, a trade has a buyer and a seller. The point is, traders usually like to know if the trade was a proactive buy or a proactive sell, or say, the taker was the buyer or the seller.

So, what makes the most sense for API users?

```
price : 1.05
amount : 2
value : 2.1
side1 (maker) : bob
side1_sold : USD
side2 (taker): alice
side2_sold : EUR
```
or
```
price : 1.05
amount : 2
value : 2.1
type: sell
side1 (maker) : bob
side2 (taker): alice
```
or
```
price : 1.05
amount : 2
value : 2.1
type: sell
seller: alice
buyer: bob
```
Actually I think the last structure makes the most sense.

But we already had `side1_account_id` and `side2_account_id`, in order to not break existing client applications, it's likely that we need to keep them. That said, we can't go with the most optimal option, but have to go with a less optimal option.

By the way, since the API was designed to return human-readable info, IMO `account_id` is not that human-readable, and `side1` and `side2` are not clear either. There is another API `get_fill_order_history` which returns more accurate data but is much less human-readable, for applications that don't output results got from API directly.

So what's the second most optional option? Probably this (with redundant data)?
```
price : 1.05
amount : 2
value : 2.1
type: sell
seller: alice
buyer: bob
side1_account_id: bob_id
side2_account_id: alice_id
```

Reply:
It is far away from what was requested initially, i could only do it by asking you more questions on how to do, more examples, etc. That is is asking for troubles(be treated like an idiot). So, again, i am going to leave it to you(the god) to do it as you want and close the issue from my side again. No hard feeling, thank you.

Reply:
Sorry for making you feeling hard. It was not my intention. My last comment was open-ended with a question, but not demanding or requesting something (of course I'm NOT the god). All the possible options I have seen or have thought of, have their own pros and cons, so I'd like to know your opinions about them.

Reply:
This was discussed and solved in private.
The format selected by now is the one that only adds `type` as adding more accounts is confusing for the end user. Re-re opening the PR, sorry about that, will try to keep the github clean of non technical discussion.